### PR TITLE
Fix login form logic

### PR DESCRIPTION
**Pull request by YaronMiro**
#54
login

### CODE REVIEW

**Comment by YaronMiro** [comment]
@amitaibu 
Ready for review.
Thanks.

File: src/elm/Login.elm [hunk 1]
```diff
@@ -176,15 +176,13 @@ view address model =
     modelForm =
       model.loginForm
 
-    disabledButton =
-      String.isEmpty modelForm.name || String.isEmpty modelForm.pass || model.status == Fetching || model.status == Fetched
+    isStatus = model.status == Fetching || model.status == Fetched
```
Comment (amitaibu) [inline src/elm/Login.elm]: isStatus => isFetchStatus

File: src/elm/Login.elm [hunk 2]
```diff
@@ -176,15 +176,13 @@ view address model =
...
 
     loginText =
-      if disabledButton
+      if isStatus && not (String.isEmpty modelForm.name && String.isEmpty modelForm.pass)
```
Comment (amitaibu) [inline src/elm/Login.elm]: make `String.isEmpty modelForm.name && String.isEmpty modelForm.pass` also a var

```elm
isFormEmpty = 
  String.isEmpty modelForm.name && String.isEmpty modelForm.pass
```

Comment (amitaibu) [inline src/elm/Login.elm]: so eventual you will have `isFetchStatus || isFormEmpty` - which is very readable.

**Comment by YaronMiro** [comment]
@amitaibu 
Ready for review.

I did some more changes:
* The inputs fields are now disabled after clicking on the login button - You were able to change their values  while the request was been made it was weird... ;) 

* Login button color is now gray when disabled - make more scene.

* I removed the variable `loginText` that was holding the "content markup" for the login button and moved all of it to the markup declaration area so know it's more readable and understood.

This is the new video after the changes:
ezgif com-crop

File: src/elm/Login.elm [hunk 3]
```diff
@@ -240,9 +239,13 @@ view address model =
             , button
               [ onClick address SubmitForm
               , class "btn btn-lg btn-primary btn-block"
-              , disabled disabledButton
+              , disabled (isFetchStatus || isFormEmpty)
+              ]
+              [ span
```
Comment (amitaibu) [inline src/elm/Login.elm]:
```elm
let
  spinner = 
    i [ class "fa fa-spinner fa-spin" ] []
in
```

```elm
[ span [ hidden <| not isFetchStatus] [ spinner ]
, span [ hidden isFetchStatus ] [ text "Login" ]
]
```

File: src/elm/Login.elm [hunk 4]
```diff
@@ -240,9 +239,13 @@ view address model =
...
+                [ hidden (not (isFetchStatus)) ]
```
Comment (amitaibu) [inline src/elm/Login.elm]: `[ hidden (not (isFetchStatus)) ] == [ hidden (not isFetchStatus) ] == [ hidden <| not isFetchStatus ] ==  [ not isFetchStatus |> hidden  ]`

**Comment by YaronMiro** [comment]
@amitaibu 
Ready for review.

**Comment by amitaibu** [comment]
Merged, thanks!

**Pull request closed without merging**